pagopa/io-dev-api-server
# [IOCOM-440] F24 PDF download for PN message

⚠️ This PR depends on #296 ⚠️

## Short description
This PR enabled the download of F24 files for PN messages (no polling yet).

## List of changes proposed in this pull request
- updated definitions
- `fp-ts` and `io-ts` upgraded to latest versions in order to use new methods
- `src/config`: path composing switched from manual to using the path library
- `src/features/pn/payloads/messages`: added F24 metadata creation on PN message
- `src/features/pn/types/messageTemplate`: type extracted from `src/types/config`
- `src/features/pn/types/messageTemplateWrapper`: type extracted from `src/config`
- `src/payloads/message`: third party attachment creation refactored in order to use the attachment's relative path instead of the attachment's id in the `url` field
- `src/persistence/mesages`: better name for single message retrieval and refactoring to use Option instead of undefined as return type
- `src/populate-persistence`: modified the attachment count parameter to reflect the new spec
- `src/routers/message`: handlers modified to use piping instead of imperative code flow
- `src/types/config`: PN types moved in their own files
- `src/utils/file`: `sendFile` renamed to `sendFileFromRootPath` to better communicate the need of using relative paths. `readBinaryFileSegment` and `isPDFFile` edited to use Either instead of throwing Errors

## How to test
Run the server instance with a PN configuration that generates both attachments and F24 and a standard configuration that generates messages with attachments. Query the `third-party-messages/:messageId/attachments/:attachmentUrl` endpoint and check that the proper data are returned.
